code/datums/components/curse_of_hunger.dm

```diff
@@ -0,0 +1,156 @@
+
+/datum/component/curse_of_hunger/proc/the_curse_ends(mob/uncursed)
+	var/obj/item/at_least_item = parent
+	STOP_PROCESSING(SSobj, src)
+	REMOVE_TRAIT(parent, TRAIT_NODROP, CURSED_ITEM_TRAIT(parent.type))
+	REMOVE_TRAIT(uncursed, TRAIT_CLUMSY, CURSED_ITEM_TRAIT(at_least_item.type))
+	REMOVE_TRAIT(uncursed, TRAIT_PACIFISM, CURSED_ITEM_TRAIT(at_least_item.type))
+	//remove either one of the signals that could have called this proc
+	UnregisterSignal(parent, list(COMSIG_ITEM_POST_UNEQUIP, COMSIG_ITEM_DROPPED))
+
+	var/turf/vomit_turf = get_turf(at_least_item)
+	playsound(vomit_turf, 'sound/effects/splat.ogg', 50, TRUE)
+	new /obj/effect/decal/cleanable/vomit(vomit_turf)
+
+	if(!add_dropdel) //gives a head start for the person to get away from the cursed item before it begins hunting again!
+		addtimer(CALLBACK(src, .proc/seek_new_target), 10 SECONDS)
+
+///proc called after a timer to awaken the AI in the cursed item if it doesn't have a target already.
+/datum/component/curse_of_hunger/proc/seek_new_target()
+	var/obj/item/cursed_item = parent
+	if(iscarbon(cursed_item.loc))
+		return
+	else if(!isturf(cursed_item.loc))
+		cursed_item.forceMove(get_turf(cursed_item))
+	//only taking the most reasonable slot is fine since it unequips what is there to equip itself.
+	cursed_item.AddElement(/datum/element/cursed, cursed_item.slot_equipment_priority[1])
+	cursed_item.visible_message("<span class='warning'>[cursed_item] begins to move on [cursed_item.p_their()] own...</span>")
+
+/datum/component/curse_of_hunger/process(delta_time)
+	var/obj/item/cursed_item = parent
+	var/mob/living/carbon/cursed = cursed_item.loc
+	///check hp
+	if(!poison_food_tolerance)
+		cursed.dropItemToGround(cursed_item, TRUE)
+		return
+	hunger++
+	if((hunger <= HUNGER_THRESHOLD_TRY_EATING) && prob(20))
+		return
+	//check hungry enough to eat something!
+	for(var/obj/item/food in cursed.contents)
+		if(!IS_EDIBLE(food))
+			continue
+		food.forceMove(cursed.loc)
+		playsound(cursed_item, 'sound/items/eatfood.ogg', 20, TRUE)
+		///poisoned food damages it
+		if(istype(food, /obj/item/food/badrecipe))
+			to_chat(cursed, "<span class='warning'>[cursed_item] eats your [food] to sate [cursed_item.p_their()] hunger, and looks [pick("queasy", "sick", "iffy", "unwell")] afterwards!</span>")
+			poison_food_tolerance--
+		else
+			to_chat(cursed, "<span class='notice'>[cursed_item] eats your [food] to sate [cursed_item.p_their()] hunger.</span>")
+		cursed.temporarilyRemoveItemFromInventory(food, force = TRUE)
+		qdel(food)
+		hunger = 0
+		return
+	///no food found: it bites you and regains some poison food tolerance
+	playsound(cursed_item, 'sound/items/eatfood.ogg', 20, TRUE)
+	to_chat(cursed, "<span class='userdanger'>[cursed_item] bites you to sate [cursed_item.p_their()] hunger!</span>")
+	var/affecting = cursed.get_bodypart(BODY_ZONE_CHEST)
+	cursed.apply_damage(60, BRUTE, affecting)
+	hunger = 0
+	poison_food_tolerance = min(poison_food_tolerance += 1, FULL_HEALTH)
```

Comment:
no this is still bad arm

You're using this in a `min` expression--you need to move the `+= 1` *out* of the `min` expression so that it is just working on basic values. (Though I am never one to suggest `++` over `+= 1`).

Reply:
thank you